bindings/python/setup.py

```diff
@@ -41,6 +41,38 @@ def get_tag(self):
 with open(version_file) as fp:
     exec(fp.read(), version)
 
+# Print a warning if the embedded libmongocrypt binary is not found.
+linux = False
+_base = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)), 'pymongocrypt')
+if sys.platform == 'win32':
+    _path = os.path.join(_base, 'mongocrypt.dll')
+elif sys.platform == 'darwin':
+    _path = os.path.join(_base, 'libmongocrypt.dylib')
+else:
+    _path = os.path.join(_base, 'libmongocrypt.so')
+    linux = True
+
+_PYMONGOCRYPT_LIB = os.environ.get('PYMONGOCRYPT_LIB')
+
+message = """
+*****************************************************\n
+The embedded libmongocrypt binary is not present (%r)\n%s
+You may need to install libmongocrypt manually and set
+the PYMONGOCRYPT_LIB environment variable.\n
+*****************************************************\n
+"""
+
+if linux:
+    pip_message = ('Please upgrade to pip>=19 to support manylinux2010 wheels'
+                   'and try again.')
+else:
+    pip_message = ''
+
+if not os.path.isfile(_path) and not (
+        _PYMONGOCRYPT_LIB and os.path.isfile(_PYMONGOCRYPT_LIB)):
+    sys.stdout.write(message % (_path, pip_message))
```

Comment:
Hmm looks like this message gets suppressed when installing via pip, even when installing from source with pip 9.0.0:
```
 pip install ./dist/pymongocrypt-0.1b2.dev0.tar.gz
Processing ./dist/pymongocrypt-0.1b2.dev0.tar.gz
Requirement already satisfied: cffi<2,>=1.12.0 in ./venv3.7-pymongocrypt/lib/python3.7/site-packages (from pymongocrypt==0.1b2.dev0)
Requirement already satisfied: cryptography<3,>=2.0 in ./venv3.7-pymongocrypt/lib/python3.7/site-packages (from pymongocrypt==0.1b2.dev0)
Requirement already satisfied: pycparser in ./venv3.7-pymongocrypt/lib/python3.7/site-packages (from cffi<2,>=1.12.0->pymongocrypt==0.1b2.dev0)
Requirement already satisfied: six>=1.4.1 in ./venv3.7-pymongocrypt/lib/python3.7/site-packages (from cryptography<3,>=2.0->pymongocrypt==0.1b2.dev0)
Requirement already satisfied: asn1crypto>=0.21.0 in ./venv3.7-pymongocrypt/lib/python3.7/site-packages (from cryptography<3,>=2.0->pymongocrypt==0.1b2.dev0)
Building wheels for collected packages: pymongocrypt
  Running setup.py bdist_wheel for pymongocrypt ... done
  Stored in directory: /Users/shane/Library/Caches/pip/wheels/99/83/b6/47f770caec2f2056e79fb29bff6a6356b36ef770f870dc3b4b
Successfully built pymongocrypt
Installing collected packages: pymongocrypt
Successfully installed pymongocrypt-0.1b2.dev0
You are using pip version 9.0.0, however version 19.2.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

It only seems to be printed when running setup.py manually (like sdist, test, bdist_wheel, etc..). I'm not sure this message is worthwhile if it can't be displayed during a normal pip install.

Reply:
Apparently pip suppresses output for the reasons described here: https://github.com/pypa/pip/issues/2933#issuecomment-290407667

That thread has a work-around but it requires verbose mode (`install -v`) for the message to be displayed. I vote we not attempt to log anything during pip install and simply document the install requirements in the README.